### ISSUE TITLE
Bug: enums in typescript triggers no-unused-vars even when they are being used.

### ISSUE THREAD

**Issue by andyanday33** [eslint/eslint#16139]
### Environment

Node version: v16.15.0
npm version: v8.5.5
Local ESLint version: v8.18.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.3.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
'env': {
  'browser': true,
  'es2021': true,
},
'extends': [
  'plugin:react/recommended',
  'google',
],
'parser': '@typescript-eslint/parser',
'parserOptions': {
  'ecmaFeatures': {
    'jsx': true,
  },
  'ecmaVersion': 'latest',
  'sourceType': 'module',
},
'plugins': [
  'react',
  '@typescript-eslint',
],
'rules': {
},
```
</details>

```ts
// This triggers no-unused-vars
enum directions {
  North = 'NORTH',
  West = 'WEST',
  East = 'EAST',
  South = 'SOUTH',
}

const demo = directions.North;
console.log(demo);
```

### What did you expect to happen?

Since the 'directions' is being used, I think it shouldn't be triggering no-unused-vars.

### What actually happened?

'directions' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
'North' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
'South' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
'West' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
'East' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @andyanday33, thanks for the issue!

The core `no-unused-vars` rule does not support TypeScript code. You should use [`@typescript-eslint/no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars/) rule instead of the core rule.

```json
{
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"]
  }  
}
```
